GCTC-NTGC/gc-digital-talent
# Feature/3219 iap analytics2

REsolves #3219 
Since the below branch became stale before Hydrogen merge .. it was seriously broken with dependencies and had to create  this pr now. No functionality changed. 
https://github.com/GCTC-NTGC/gc-digital-talent/pull/3675 

### How to Test Local
Check out this branch
Add "InstrumentationKey=8f4f1fd8-8831-4a3b-84a0-20245c86e418;IngestionEndpoint=https://canadacentral-1.in.applicationinsights.azure.com/;LiveEndpoint=https://canadacentral.livediagnostics.monitor.azure.com/" as the value for APPLICATIONINSIGHTS_CONNECTION_STRING in .env
Open the network tab in your browser
Navigate to '/en/indigenous-it-apprentice'
Search for track in the network tab and verify the headers and payload 

### How to verify the metrics in  Azure 

- Deploy the branch in dev 
- Add the env variable a above 
- Navigate to IAP home page and make some requests
- login to chrome with your pc account and go to portal.azure.com
- click "AppInsights"
- click "Transaction Search" and look for the hits for the time you hit the page 
![image (3)](https://user-images.githubusercontent.com/89482910/189935439-b423b3c1-a416-410a-a7c6-62b50e622d34.png)


### Deployment Notes

**Dev & UAT**
Add the below environment variable
APPLICATIONINSIGHTS_CONNECTION_STRING=""

( empty string as we don't want to collect metrics from these environments )

**Prod**
APPLICATIONINSIGHTS_CONNECTION_STRING= "InstrumentationKey=8f4f1fd8-8831-4a3b-84a0-20245c86e418;IngestionEndpoint=https://canadacentral-1.in.applicationinsights.azure.com/;LiveEndpoint=https://canadacentral.livediagnostics.monitor.azure.com/"
